torchq/qtensor.py

```diff
@@ -0,0 +1,61 @@
+from typing import List
+
+import numpy as np
+import qutip as qt
+import torch
+
+
+class QTensor(torch.Tensor):
+    """
+    A torch tensor that represents a quantum object (operator, density matrix, state vector...)
+
+    Subclasses of Tensor can slow torch down significantly
+    TODO: check performance impact and implement mitigations
+    Ref : https://benjaminwarner.dev/2022/06/14/debugging-pytorch-performance-decrease
+    """
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Should we provide an optional `type` argument as in qutip? This would be `{'bra', 'ket', 'oper', 'operator-ket', 'operator-bra', 'super'}`.

Reply:
I've never used them. How useful is it?
If we put them, shouldn't we need a `density-matrix` type?

Reply:
Not sure what it's for to be honest! I guess if we ever need it we can add it then.